python/ray/serve/pipeline/generate.py

```diff
@@ -124,29 +124,36 @@ def transform_ray_dag_to_serve_dag(dag_node):
 
 def extract_deployments_from_serve_dag(
     serve_dag_root: DAGNode,
-) -> List[Deployment]:
+) -> Tuple[List[Deployment], List[Deployment]]:
     """Extract deployment python objects from a transformed serve DAG. Should
     only be called after `transform_ray_dag_to_serve_dag`, otherwise nothing
     to return.
 
     Args:
         serve_dag_root (DAGNode): Transformed serve dag root node.
     Returns:
-        List[Deployment]: List of deployment python objects fetched from serve
-            dag.
+        body_deployments (List[Deployment]): List of deployment python objects
+            fetched from serve dag.
+        root_deployment (List[Deployment]): Root deployment that is exposed to
+            user via http. Might be empty if user has a ray dag that uses a
+            function or class method node as entrypoint.
     """
-    deployments = {}
+    body_deployments = {}
+    root_deployment = []
```

Comment:
Can there ever be more than one `root_deployment`? If not, can we change this to either be `None` or that deployment (i.e. make type `Optional[Deployment]` instead of `List[Deployment]`)?

Reply:
yep there could only be one. But setting it as a single variable `None` didn't work so well with the `extractor` below compare to collection .. i didn't polish it much tho, let me give it another try and see.